code/app/src/main/java/com/example/goldencarrot/data/db/UserRepository.java

```diff
@@ -203,6 +203,36 @@ public void getSingleUser(String androidId, FirestoreCallbackSingleUser callback
                     Log.e(TAG, "Error fetching user: " + e.getMessage(), e);
                 });
     }
+    /**
+     * Retrieves the current user's ID based on their Android ID.
+     *
+     * @param androidId The unique Android ID of the user.
+     * @param callback A callback to handle the result (userId if exists, or null if not found).
+     */
+    public void getCurrentUserId(String androidId, FirestoreCallbackUserId callback) {
```

Comment:
the Android Id and the User Id are the same thing

this method is getting an id using the same id

please remove this


there is an example on how to do this in MainActivity.java

Reply:
updated!